dotnet/src/Connectors/Connectors.UnitTests/OpenAI/TextToImage/AzureOpenAITextToImageTests.cs

```diff
@@ -13,17 +14,60 @@
 namespace SemanticKernel.Connectors.UnitTests.OpenAI.TextToImage;
 
 /// <summary>
-/// Unit tests for <see cref="AzureOpenAITextToImageTests"/> class.
+/// Unit tests for <see cref="AzureOpenAITextToImageServiceTests"/> class.
 /// </summary>
-public sealed class AzureOpenAITextToImageTests
+public sealed class AzureOpenAITextToImageServiceTests
 {
+    [Fact]
+    public void ItValidatesTheModelId()
+    {
+        // Arrange
+        const string InvalidModel = "dall-e-3";
+
+        const string ValidModel1 = "dall-e-2";
+        const string ValidModel2 = "Dall-E-2";
+        const string ValidModel3 = "";
+
+        // Act
+        Assert.Throws<NotSupportedException>(() => new AzureOpenAITextToImageService(modelId: InvalidModel, endpoint: "https://az.com", apiKey: "abc"));
+
+        // No exceptions for these
+        var _ = new AzureOpenAITextToImageService(modelId: ValidModel1, endpoint: "https://az.com", apiKey: "abc");
+        _ = new AzureOpenAITextToImageService(modelId: ValidModel2, endpoint: "https://az.com", apiKey: "abc");
+        _ = new AzureOpenAITextToImageService(modelId: ValidModel3, endpoint: "https://az.com", apiKey: "abc");
+    }
```

Comment:
This is a good example for `[Theory]`
```suggestion
    [Theory]
    [InlineData("dall-e-3", false)]
    [InlineData("dall-e-2", true)]
    [InlineData("Dall-E-2", true)]
    [InlineData("", true)]
    public void ItValidatesTheModelId(modelId, shouldBeValid)
    {
        // Act
        if (!shouldBeValid) 
        {
            Assert.Throws<NotSupportedException>(() => new AzureOpenAITextToImageService(modelId: modelId, endpoint: "https://az.com", apiKey: "abc"));
        }
        else 
        {
            // No exceptions for these
            var _ = new AzureOpenAITextToImageService(modelId: modelId, endpoint: "https://az.com", apiKey: "abc");
        }
    }
```